chef/cookbooks/provisioner/recipes/base.rb

```diff
@@ -328,6 +326,16 @@
       if current_priority != attrs[:priority]
         `zypper --non-interactive modifyrepo --priority #{attrs[:priority]} #{name}`
       end
+      # check if attrs contains the :purge key with the databag name as a value
+      if attrs[:purge]
+        repo_bag = attrs[:purge]
+
+        Chef::Log.info("Purging no longer used #{name} zypper repository...")
+        `zypper --non-interactive removerepo #{name}`
+        Chef::Log.info("Purging repository data bag #{repo_bag}...")
+        # dont rescue here because we want to exit the recipe in case of an error
+        Chef::DataBag.chef_server_rest.delete_rest("data/#{repo_bag}")
```

Comment:
can't do that yet, because another node might still not have run the zypper removerepo...
also it seems to forbidden to delete a databag from a client (non admin) node